tests/MongoDB.Driver.Tests/Specifications/unified-test-format/UnifiedTestOperations/UnifiedAggregateCollectionOperation.cs

```diff
@@ -0,0 +1,108 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MongoDB.Bson;
+
+namespace MongoDB.Driver.Tests.Specifications.unified_test_format.UnifiedTestOperations
+{
+    public class UnifiedAggregateCollectionOperation : IUnifiedTestOperation
+    {
+        private IMongoCollection<BsonDocument> _collection;
+        private AggregateOptions _options;
+        private PipelineDefinition<BsonDocument, BsonDocument> _pipeline;
+
+        public UnifiedAggregateCollectionOperation(
```

Comment:
The new implementation of test operations has a narrowed dependency list to represent dependencies more clear, reduce the load on operation classes logic and reduce the code not directly responsible for operation execution.